include/pybind11/cast.h

```diff
@@ -1,3 +1,7 @@
+// clang-format off
+// Please consider `clang-format`ting this file before starting with significant
```

Comment:
Objective is this statement isn't completely clear (could be made more concise).

Is this a "DNM" (Do Not Merge) for this PR, or is this something that would get merged in mainline?
If it's draft for this PR, consider reflecting that?
```
// DNM: This will be reformatted as part of this PR.
```

If it's intended to be merged to mainline, consider:
```
// Please consider `clang-format`ting any new lines added when submitting a PR.
```

However, TBH, I'm not sure how smooth of a process that will be?

Reply:
What I had in mind is explained in CONTRIBUTING.md. I was hoping people look there as suggested by the `See also` in the next line here. This text from CONTRIBUTING.md seemed too long to duplicate at the top of all files:

> Please consider `clang-format`ing these files before starting with major changes, ideally in a separate commit, or even a separate PR, to keep development changes separate from `clang-format` cleanup.

Formatting only new or modified lines is another option, but I didn't want to make the explanations too complicated, and I'm also ambivalent about the resulting patchwork of `// clang-format on/off` directives.

I think the healthiest option is to globally clang-format everything in one big PR that does nothing else, but I created this PR as a minimalistic alternative because Wenzel didn't seem supportive. Wenzel doesn't even seem to support this one.